core/src/main/java/com/linecorp/armeria/server/annotation/JacksonRequestConverterFunction.java

```diff
@@ -66,8 +67,13 @@ public Object convertRequest(ServiceRequestContext ctx, AggregatedHttpMessage re
                                     contentType.subtype().endsWith("+json"))) {
             final ObjectReader reader = readers.computeIfAbsent(expectedResultType, mapper::readerFor);
             if (reader != null) {
-                return reader.readValue(request.content().toString(
-                        contentType.charset().orElse(StandardCharsets.UTF_8)));
+                final String content = request.content().toString(
+                        contentType.charset().orElse(StandardCharsets.UTF_8));
+                try {
+                    return reader.readValue(content);
+                } catch (JsonProcessingException e) {
+                    throw new IllegalArgumentException("failed to parse a JSON document: " + content, e);
```

Comment:
I'm somewhat concerned about stringifying `content` because it can be very long.

Reply:
Alright let me just remove the `content`. Thank you~!!